gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceStubSettingsClassComposer.java

```diff
@@ -370,6 +370,16 @@ protected MethodDefinition createApiClientHeaderProviderBuilderMethod(
             .setReturnType(returnType)
             .build();
 
+    if (!Strings.isNullOrEmpty(service.apiVersion())) {
```

Comment:
I know part 1 is not merged yet. Adding a note here to remind us for `service.hasApiVersion()`